c/tskit/tables.c

```diff
@@ -2542,8 +2542,14 @@ tsk_mutation_table_append_columns(tsk_mutation_table_t *self, tsk_size_t num_row
     } else {
         memcpy(self->parent + self->num_rows, parent, num_rows * sizeof(tsk_id_t));
     }
-    memcpy(self->time + self->num_rows, time, num_rows * sizeof(double));
-
+    if (time == NULL) {
+        /* If time is NULL, set all times to NAN which is the default */
```

Comment:
We need to be careful here - need to make sure it's a non-signalling NaN, or else weird things will happen on different platforms.

Reply:
My understanding here from https://en.wikipedia.org/wiki/IEEE_754 is that anything with exponent == 0xFF and fraction != 0 is a NaN and that the highest fraction bit (bit 22) is `1` for a quiet NaN and `0` for signalling. I think the best thing to do is put in some asserts (hopefully compile time) that 0xFF*8 is a non signalling NaN.

Reply:
We could do a loop here and assign manually to a NAN macro? Might be simplest in this case. We can use the NAN value defined in GNU C, and make sure it's defined otherwise.